### DISCUSSION TITLE
Problem when submitting an app to the AppStore

### DISCUSSION THREAD

**Discussion by Dmitriy-Holovnia** [codemagic-ci-cd/codemagic-docs#2484] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

I am trying to push app to AppStore. My app have two targets. Default and NotificationServiceExtension. I have two profiles for two targets. When I trying to push the app I receiving error:

2024-04-18 12:15:55.349 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90085) No architectures in the binary. Lipo failed to detect any architectures in the bundle executable. (ID: f319831f-4d7f-479b-a067-71fc2553b56e)
2024-04-18 12:16:05.884  INFO: [ContentDelivery.Uploader] 
=============
UPLOAD FAILED with 1 error.
=============
{"tool-version":"7.006.15006","tool-path":"\/Applications\/Xcode-15.3.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"14.3.1","product-errors":[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=No architectures in the binary. Lipo failed to detect any architectures in the bundle executable., id=f319831f-4d7f-479b-a067-71fc2553b56e, code=STATE_ERROR.VALIDATION_ERROR.90085, title=Asset validation failed, NSLocalizedFailureReason=No architectures in the binary. Lipo failed to detect any architectures in the bundle executable., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90085","NSLocalizedFailureReason":"No architectures in the binary. Lipo failed to detect any architectures in the bundle executable. (ID: f319831f-4d7f-479b-a067-71fc2553b56e)"},"code":90085}]}

### Expected results

Success push

### Actual results

2024-04-18 12:15:55.349 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90085) No architectures in the binary. Lipo failed to detect any architectures in the bundle executable. (ID: f319831f-4d7f-479b-a067-71fc2553b56e)
2024-04-18 12:16:05.884  INFO: [ContentDelivery.Uploader] 
=============
UPLOAD FAILED with 1 error.
=============
{"tool-version":"7.006.15006","tool-path":"\/Applications\/Xcode-15.3.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"14.3.1","product-errors":[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=No architectures in the binary. Lipo failed to detect any architectures in the bundle executable., id=f319831f-4d7f-479b-a067-71fc2553b56e, code=STATE_ERROR.VALIDATION_ERROR.90085, title=Asset validation failed, NSLocalizedFailureReason=No architectures in the binary. Lipo failed to detect any architectures in the bundle executable., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90085","NSLocalizedFailureReason":"No architectures in the binary. Lipo failed to detect any architectures in the bundle executable. (ID: f319831f-4d7f-479b-a067-71fc2553b56e)"},"code":90085}]}

### Build id (optional)

66210e8c5c14456c5d528ec9

**Comment by dtrdic** [comment]
Hello @Dmitriy-Holovnia can you please check [this github discussion](https://github.com/flutter/flutter/issues/101461), as the issue seems to be related

**Reply by Dmitriy-Holovnia** [reply]
`AppFrameworkInfo` is only present in flutter. I have a native iOS app

**Reply by dtrdic** [reply]
hey @Dmitriy-Holovnia i'll close this thread as i see your latest build is successful. If you are still needing assistance please reopen this discussion. Have a nice day!
